int32 agent_id
string agent_state
